src/main/kotlin/uk/gov/dluhc/printapi/service/PrintResponseFileReadinessService.kt

```diff
@@ -0,0 +1,47 @@
+package uk.gov.dluhc.printapi.service
+
+import mu.KotlinLogging
+import org.apache.commons.lang3.time.StopWatch
+import org.springframework.stereotype.Service
+import uk.gov.dluhc.printapi.config.SftpProperties
+import uk.gov.dluhc.printapi.messaging.models.ProcessPrintResponseFileMessage
+
+private val logger = KotlinLogging.logger {}
+
+@Service
+class PrintResponseFileReadinessService(
+    private val sftpService: SftpService,
+    private val sftpProperties: SftpProperties,
+    private val printMessagingService: PrintMessagingService
+) {
+
+    fun markPrintResponseFileForProcessing() {
+        val stopWatch = StopWatch.createStarted()
+        val outboundFolderPath = sftpProperties.printResponseDownloadDirectory
+        logger.info { "Finding matching print responses from directory: [$outboundFolderPath]" }
+
+        val unprocessedFiles = sftpService.identifyFilesToBeProcessed(outboundFolderPath)
+        logger.info { "Found [${unprocessedFiles.size}] unprocessed print responses" }
+
+        val filesFailedToProcess = mutableListOf<String>()
+        unprocessedFiles.forEachIndexed { index, unprocessedFileName ->
+            try {
+                sftpService.markFileForProcessing(
+                    directory = outboundFolderPath,
+                    originalFileName = unprocessedFileName
+                ).also {
+                    val messagePayload = ProcessPrintResponseFileMessage(outboundFolderPath, it)
+                    logger.info { "Submitting SQS message for file: [${index + 1} of ${unprocessedFiles.size}] with payload: $messagePayload" }
+                    printMessagingService.submitPrintResponseFileForProcessing(messagePayload)
+                }
+            } catch (e: RuntimeException) {
+                filesFailedToProcess.add(unprocessedFileName)
+                logger.warn { "Error renaming [$unprocessedFileName] due to [${e.message}]. Processing will continue for rest of the files" }
+            }
+        }
+
+        stopWatch.stop()
+        logger.warn { "${filesFailedToProcess.size} files failures that were not processed $filesFailedToProcess" }
```

Comment:
Do we need this log statement? Especially when there are zero files that failed to process? IE. in the case of zero failed files we will log a warning:
```
WARN: 0 files failures that were not processed []
```
No failures is not really a warning condition is it?

I guess we could wrap this log statement in an `if` statement, but I would ask "whats the point" when each failed filename is logged within the main loop ?

Reply:
Minor - Do we really need the stopwatch? Do we care that it took x seconds to process? If we really care because we're trying to diagnose something, wont the AWS logstream timestamps between the start and end messages tell us that?